File: proj/CoreLibrary/src/com/jasonzqshen/familyaccounting/core/masterdata/BusinessAreaMasterData.java
package com.jasonzqshen.familyaccounting.core.masterdata;

import com.jasonzqshen.familyaccounting.core.CoreDriver;
import com.jasonzqshen.familyaccounting.core.exception.NullValueNotAcceptable;
import com.jasonzqshen.familyaccounting.core.utils.CriticalLevel;

public class BusinessAreaMasterData extends MasterDataBase {
	public static final String FILE_NAME = "business.xml";

	private CriticalLevel _criticalLevel;

	/**
	 * 
	 * @param id
	 * @param descp
	 * @param parser
	 * @throws NullValueNotAcceptable
	 */
	public BusinessAreaMasterData(CoreDriver coreDriver, MasterDataManagement management, MasterDataIdentity id,
			String descp, CriticalLevel level) throws NullValueNotAcceptable {
		super(coreDriver, management, id, descp);

		_criticalLevel = level;
	}

	/**
	 * set critical level
	 * 
	 * @param l
	 *            critical level
	 * @throws NullValueNotAcceptable
	 */
	public void setCriticalLevel(CriticalLevel l) throws NullValueNotAcceptable {
		if (l == null) {
			throw new NullValueNotAcceptable("Critical Level");
		}
		this.setDirtyData();
		_criticalLevel = l;
	}

	/**
	 * get critical level
	 * 
	 * @return critical level
	 */
	public CriticalLevel getCriticalLevel() {
		return _criticalLevel;
	}

	@Override
	public String toXML() {
		String superStr = super.toXML();
		return String.format("%s %s=\"%s\"", superStr,
				MasterDataUtils.XML_CRITICAL_LEVEL, _criticalLevel);
	}
}

File: trunk/proj/WinStore_FamilyAccounting/WinStore_FamilyAccountingCore/MasterData/BusinessAreaMasterData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace WinStore_FamilyAccountingCore.MasterData
{
    public enum CriticalLevel
    {
        HIGH = 'H', MEDIUM = ('M'), LOW = ('L')
    }


    public class BusinessAreaMasterData : MasterDataBase
    {
        public static readonly String FILE_NAME = "business.xml";

        private CriticalLevel _criticalLevel;
        public CriticalLevel CriLevel { get { return _criticalLevel; } }
        /// <summary>
        /// 
        /// </summary>
        /// <param name="coreDriver"></param>
        /// <param name="management"></param>
        /// <param name="id"></param>
        /// <param name="descp"></param>
        /// <param name="level"></param>
        /// <exception cref="ArgumentNullException">argument is null</exception>
        public BusinessAreaMasterData(CoreDriver coreDriver, MasterDataManagement management, MasterDataIdentity id,
                String descp, CriticalLevel level)
            : base(coreDriver, management, id, descp)
        {
            _criticalLevel = level;
        }

        /// <summary>
        /// set critical level
        /// </summary>
        /// <param name="l"></param>
        public void SetCriticalLevel(CriticalLevel l)
        {
            this.SetDirtyData();
            _criticalLevel = l;
        }

        /// <summary>
        /// parse to XML
        /// </summary>
        /// <returns></returns>
        public override XElement ToXML()
        {
            XElement xelem = base.ToXML();
            xelem.Add(new XAttribute(MasterDataUtils.XML_CRITICAL_LEVEL
                , ((char)_criticalLevel).ToString()));
            return xelem;
        }
    }
}
